lib/scan.js

```diff
@@ -177,6 +179,9 @@ const scan = (input, options) => {
         isGlob = token.isGlob = true;
         isExtglob = token.isExtglob = true;
         finished = true;
+        if (code === CHAR_EXCLAMATION_MARK && index === start) {
+          negatedExtglob = true;
+        }
```

Comment:
Nice work, this comparison was the first thing I was going to check when I saw the PR. 👍🏻